managers/BaseStatusPoller.js

```diff
@@ -0,0 +1,134 @@
+            const currentAnnotations = metadata.annotations;
+            const patchAnnotations = currentAnnotations ? currentAnnotations : {};
+            patchAnnotations.lockedByTaskPoller = JSON.stringify({
+              lockTime: new Date(),
+              ip: config.broker_ip
+            });
+            metadata.annotations = patchAnnotations;
+            // Handle conflict also
+            return eventmesh.apiServerClient.updateResource({
+                resourceGroup: resourceDetails.resourceGroup,
+                resourceType: resourceDetails.resourceType,
+                resourceId: metadata.name,
+                metadata: metadata
+              })
+              .tap((updatedResource) => logger.debug(`Successfully acquired status poller lock for resource ${this.resourceType} with options: ${JSON.stringify(options)}\n` +
+                `Updated resource with poller annotations is: `, updatedResource))
+              .then(() => {
+                if (!_.includes(this.validStateList, _.get(resourceBody, 'status.state'))) {
+                  this.clearPoller(metadata.name, intervalId);
+                } else {
+                  return this.getStatus(resourceBody, intervalId);
+                }
+              })
+              .catch(Conflict, () => {
+                logger.debug(`Not able to acquire status poller processing lock for resource ${object.metadata.name}, Request is probably picked by other worker`);
+              });
+          }
+        });
+      })
+      .catch(err => {
+        logger.error(`Error occured while polling status for resource: ${this.resourceType}, id: ${object.metadata.name}, clearing poller interval now`, err);
+        this.clearPoller(object.metadata.name, intervalId);
+      });
+
+  }
+
+  startPoller(event) {
+    logger.debug(`Received ${this.resourceType} Event:`, event);
+    const resourceId = event.object.metadata.name;
+    if (_.includes(this.validEventList, event.type) && !this.pollers[resourceId]) {
+      logger.debug(`Starting status poller for resource type ${this.resourceType}, id:`, resourceId);
+      // Poller time should be little less than watch refresh interval as 
+      const intervalId = setInterval(() => this.pollStatus(event.object, intervalId), this.pollInterval);
+      this.pollers[resourceId] = intervalId;
+    }
+  }
+
+  init() {
+    const queryString = `state in (${_.join(this.validStateList, ',')})`;
+    return eventmesh.apiServerClient.registerWatcher(this.resourceGroup, this.resourceType, this.startPoller.bind(this), queryString)
+      .then(stream => {
+        logger.debug(`Successfully set watcher on ${this.resourceType} resource types for status polling with query string:`, queryString);
+        return Promise
+          .delay(CONST.APISERVER.POLLER_WATCHER_REFRESH_INTERVAL)
+          .then(() => {
+            logger.debug(`Refreshing stream after ${CONST.APISERVER.POLLER_WATCHER_REFRESH_INTERVAL}`);
+            stream.abort();
```

Comment:
why stream abort required? just for understanding.

Reply:
We have to close the old stream (if it hasn't died) in order to avoid multiple connections